### DISCUSSION TITLE
How to search through multiple history stores (in various directories on one machine)

### DISCUSSION THREAD

**Discussion by halloleo** [xonsh/xonsh#5246] [Q&A]
I backup the xonsh folders `~/.local/share/xonsh/` from multiple machines to directroies on my main machine.

So I have a bunch of directories:

    /backup/machineA/stuff/here/.local/share/xonsh/
    /backup/machineB/other/stuff/here/.local/share/xonsh/
    /backup/machineC/more/here/.local/share/xonsh/
   ~/.local/share/xonsh/

Now I would like to search _all_ these histories _together_ via something like 

    history show all | grep "findme"`

How can I do this?

Thanks heaps for any pointers!

**Comment by anki-code** [comment]
Two ways:

1. Convert history [from json to sqlite](https://github.com/xonsh/xonsh/discussions/4132) and [do everything](https://github.com/xonsh/xonsh/issues/2429#issuecomment-1235212003).

2. As I remember [the name of json is session ID and it's UUID](https://github.com/xonsh/xonsh/blob/919bedd190909843bbb2c492f5cd374810076ef2/xonsh/history/json.py#L402-L403) so you can also just copy all files in one directory, reload xonsh (may be by setting [`$XONSH_DATA_DIR`](https://xon.sh/envvars.html#xonsh-data-dir) before) and try again.

**Reply by halloleo** [reply]
Thanks @anki-code. Great info!

Converting to sqlite is a too big step for me at the moment, so I will try way 2 and report back here.

**Reply by anki-code** [reply]
Also by passing across 1 you can find the code to read json files.
